File: JavaScript/Level.2/프린터.js
function solution(priorities, location) {
    var answer = 0;
    while (true) {
        let priority = priorities[0];
        priorities.shift();
        if (priorities.some((e) => e > priority)) {
            priorities.push(priority);
            if (location === 0) location = priorities.length;
        } else {
            ++answer;
            if (location === 0) break;
        }
        --location;
    }
    return answer;
}

File: C++/Level.2/프린터.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
   	while (1) {
    	int priority = priorities.front();
        int n = priorities.size();
        int max = *max_element(priorities.begin(), priorities.end());
        if (max == priority) {
        	++answer;
            if (location == 0) break;
            priorities.erase(priorities.begin());
            --location;
        } else {
            priorities.push_back(priority);
            priorities.erase(priorities.begin());
            if (location == 0) {
                location = n - 1;
            } else {
                --location;
            } 
        }
    } 
    return answer;
}